### ISSUE TITLE
Better "Save" function for query editor

### ISSUE THREAD

**Issue by alexbohr** [sequelpro/sequelpro#2239]
I use Sequel Pro 99.9% as a Query editor, so I might spend 30 minutes typing SQL into the editor window than hit close by accident and there's no prompt "Save Changes Before Exiting?"

Also, there's not even a "Save" option, only "Save As" which requires me to go into a window and select the file I want to save as and make sure I'm not over-writing the wrong scratch-file in my scratch directory.

So basically: Please treat the query editor window as a document, let me save changes with a single click and prompt for save changes before exit.

Thanks!

**Comment by delphinian** [comment]
To add more, when I open a query,and then delete the contents of the query window to start a new query. When I save the query, it will over-write the contents of the one that I opened previously. Even though this is a new and different query as the last one.  Adding a SAVE-AS would reduce over writes.

**Comment by cristiroma** [comment]
:+1: I would suggest even a better approach, if I close the application (by mistake or it crashes) to restore where it left off - restoring the state where it left off.

To me this it would be a killer feature

**Comment by dmoagx** [comment]
@cristiroma That is #304

**Comment by joshdavenport** [comment]
Yeah, have to reiterate the point made by @delphinian here. When building multiple queries consecutively that are wanted to be saved in separate .sql files, there's no way to get rid of the save context. After hitting save for the first time, any save will save to the same file, overwriting the original. No way around it short of closing the entire tab/connection and reopening.

Didn't realise this is how it worked at first and lost a built query from assuming that the save menu item would bring up a dialog to save. I suppose that's how you'd expect an "Export" menu item to behave, but somehow it felt like that's how the save item would work. Perhaps _because_ there was no "Save As".

**Comment by mattdoran** [comment]
Agree with all of this. Sequel Pro is fantastic.  The behavior of it as an editor is the single biggest downside.

It's too easier to overwrite changes, lose changes, etc.

It doesn't need to be a fully fledged file editor but if it could improve the following it would be massive:
- Warn on close without saving.
- Clarity on the editing state - am I editing a file? which file? is is dirty/modified?
- Notice file changes in background, and offer to reload.  e.g. if you've edited an opened query in another editor.

**Comment by atippett** [comment]
This is a request not unlike how mysqlworkbench works where it will treats a query window like a unsaved buffer so if you exist and reopen that buffer is reopened with queries on it.  Buffer is discarded only when you close the tab without saving.

**Comment by dmoagx** [comment]
@atippett What you are talking about is #304

**Comment by atippett** [comment]
@dmoagx  agreed.
